zlib/grabber.go

```diff
@@ -238,6 +238,8 @@ func makeGrabber(config *Config) func(*Conn) error {
 			config.ErrorLog.Errorf("Conversation error with remote host %s: %s",
 				c.RemoteAddr().String(), err.Error())
 		}
+
+		c.Close()
```

Comment:
From what I could tell, we were never explicitly closing any of our HTTP connections. This seems important because we are doing HTTP 1.1 in all our initial requests, which defaults to keep-alive connections.